File: Reqnroll/Chapter18/18.3-CleanStartPattern/After/WIMP.Specs/StepDefinitions/MenuAdminStepDefinitions.cs
using Reqnroll;

using WIMP.Specs.Drivers;
using WIMP.Specs.Support;

namespace WIMP.Specs.StepDefinitions;

[Binding]
public class MenuAdminStepDefinitions(MenuBackdoorDriver menuBackdoorDriver)
{
    [Given("the restaurant menu is")]
    public void GivenTheRestaurantMenuIs(MenuItemData[] menuItems)
    {
        menuBackdoorDriver.SetMenuItems(menuItems);
    }
}

File: CucumberJVM/Chapter21/21.2-ResourcePoolingPattern/Before/src/test/java/com/wimp/app/specs/stepdefinitions/MenuAdminStepDefinitions.java
package com.wimp.app.specs.stepdefinitions;

import com.wimp.app.specs.drivers.MenuBackdoorDriver;
import com.wimp.app.specs.support.MenuItemData;
import io.cucumber.java.en.Given;

import java.util.List;

public class MenuAdminStepDefinitions {
    private final MenuBackdoorDriver menuBackdoorDriver;

    public MenuAdminStepDefinitions(MenuBackdoorDriver menuBackdoorDriver) {
        this.menuBackdoorDriver = menuBackdoorDriver;
    }

    @Given("the restaurant menu is")
    public void theRestaurantMenuIs(List<MenuItemData> menuItems) { menuBackdoorDriver.setMenuItems(menuItems); }
}
